tests/Functional/ContentCallForProposalsRenderTest.php

```diff
@@ -0,0 +1,379 @@
+    $node->set('oe_reference_code', 'Reference code')->save();
+    $this->drupalGet($node->toUrl());
+
+    $details_expected_values = [
+      'items' => [
+        [
+          'label' => 'Status',
+          'body' => 'N/A',
+        ], [
+          'label' => 'Reference',
+          'body' => 'Reference code',
+        ], [
+          'label' => 'Publication date',
+          'body' => $publication_date->format('d F Y'),
+        ], [
+          'label' => 'Deadline model',
+          'body' => 'Permanent',
+        ],
+      ],
+    ];
+    $field_list_assert->assertPattern($details_expected_values, $content_items[0]->getHtml());
+
+    // Assert Publication in the official journal field.
+    $node->set('oe_call_proposals_journal', [
+      'uri' => 'http://example.com/journal',
+      'title' => 'Official Journal Reference',
+    ])->save();
+    $this->drupalGet($node->toUrl());
+
+    $details_expected_values['items'][2]['body'] = $publication_date->format('d F Y') . "\n  in\n  Official Journal Reference" . chr(194) . chr(160);
+    $field_list_assert->assertPattern($details_expected_values, $content_items[0]->getHtml());
+
+    $journal_link_selector = '//*[text() = "Publication date"]/following-sibling::dd[1]/div';
+    $publication_details = $this->assertSession()->elementExists('xpath', $journal_link_selector);
+    $this->assertLinkIcon($publication_details, 'Official Journal Reference', 'http://example.com/journal');
+
+    // Assert Opening date field.
+    $opening_date = (clone $static_time)->modify('- 3 days');
+    $node->set('oe_call_proposals_opening_date', ['value' => $opening_date->format('Y-m-d')]);
+    $node->set('oe_call_proposals_journal', NULL)->save();
+    $this->drupalGet($node->toUrl());
+
+    $header_expected_values['meta'] = 'Call for proposals | Open';
+    $assert->assertPattern($header_expected_values, $page_header->getOuterHtml());
+
+    $details_expected_values = [
+      'items' => [
+        [
+          'label' => 'Status',
+          'body' => 'Open',
+        ], [
+          'label' => 'Reference',
+          'body' => 'Reference code',
+        ], [
+          'label' => 'Publication date',
+          'body' => $publication_date->format('d F Y'),
+        ], [
+          'label' => 'Opening date',
+          'body' => $opening_date->format('d F Y'),
+        ], [
+          'label' => 'Deadline model',
+          'body' => 'Permanent',
+        ],
+      ],
+    ];
+    $field_list_assert->assertPattern($details_expected_values, $content_items[0]->getHtml());
+
+    // Assert Deadline model and Deadline dates fields.
+    $deadline_date1 = (clone $static_time)->modify('- 3 days');
+    $node->set('oe_call_proposals_model', 'single_stage');
+    $node->set('oe_call_proposals_deadline', $deadline_date1->format(DateTimeItemInterface::DATETIME_STORAGE_FORMAT));
+    $node->save();
+    $this->drupalGet($node->toUrl());
+
+    $deadline_date1->setTimeZone(new \DateTimeZone('Australia/Sydney'));
```

Comment:
Why do we need to set the timezone here?

Reply:
I see, the same would happen for the other dates we are checking but this is the only place where we check the hour and timezone so hence the solution. However, there is no need to actually format any date to test this, we know what the result needs to be (we froze the time for a reason) so instead of formatting any date we should simply pass the string as the expected result (we do it like this consistently across all the other tests

Reply:
Hardcoded dates are used now.